### PR TITLE
Min/Max diameter with mapping

### PR DESCRIPTION
**Pull request by luca-heltai**

### CODE REVIEW

File: source/grid/grid_tools.cc [hunk 1]
```diff
@@ -2770,15 +2770,43 @@ namespace GridTools
 
 
 
+  namespace
+  {
+    template <int dim, int spacedim>
+    double diameter(const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+                    const Mapping<dim,spacedim> &mapping)
+    {
+      auto vertices = mapping.get_vertices(cell);
```
Comment (tjhei) [inline source/grid/grid_tools.cc]: ``const auto &`` ?

File: tests/grid/grid_tools_min_max_diameter_01.cc [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#include "../tests.h"
```
Comment (tjhei) [inline tests/grid/grid_tools_min_max_diameter_01.cc]: would you mind adding a comment like:
```
// test GridTools::minimal_cell_diameter and GridTools::maximal_cell_diameter with a mapping
```

File: source/grid/grid_tools.cc (hunk 1 above)
Comment (luca-heltai) [reply source/grid/grid_tools.cc]: cannot be a reference. I'll add the const, though. :)

File: tests/grid/grid_tools_min_max_diameter_01.cc [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <vector>
+#include <string>
+#include <sstream>
+
+#define PRECISION 2
+
+// test GridTools::minimal_cell_diameter and GridTools::maximal_cell_diameter with a mapping
```
Comment (tjhei) [inline tests/grid/grid_tools_min_max_diameter_01.cc]: did you put the comment down there on purpose? we typically put it above the includes

**Comment by luca-heltai** [comment]
When this gets merged, I'll go forward with #6294. Merge?

**Pull request merged**